core/templates/dev/head/pages/collection_editor/settings_tab/CollectionPermissionsCardDirective.js

```diff
@@ -22,10 +22,24 @@ oppia.directive('collectionPermissionsCard', [function() {
     restrict: 'E',
     templateUrl: 'inline/collection_permissions_card_directive',
     controller: [
-      '$scope',
-      function($scope) {
-        $scope.ownerNames = GLOBALS.ownerNames;
-        $scope.isPrivate = GLOBALS.isPrivate;
+      '$scope', 'CollectionRightsObjectFactory',
+      'CollectionRightsBackendApiService',
+      function($scope, CollectionRightsObjectFactory,
+        CollectionRightsBackendApiService) {
+        $scope.collectionId = GLOBALS.collectionId;
+        $scope.collectionRightsObject = null;
+        $scope.hasRightsLoaded = false;
+        $scope.isPrivate =
```

Comment:
I am thinking it might be better if the first `load` were in `CollectionEditorStateService`, so that it is guaranteed to be the same `fetch`. The problem is that when the user first goes to the settings page the (ie. when a user bookmarks that settings link and goes there first), the cache is empty and both the Navbar and the Permissions tab also go to fetch from load (since both controllers call load).